setup.py

```diff
@@ -168,7 +168,12 @@ def _main():
             "build_ext": BuildExtension.with_options(no_python_abi_suffix=True),
             "clean": clean,
         },
-        install_requires=[pytorch_package_dep, "numpy", "packaging"],
+        install_requires=[
+            pytorch_package_dep,
+            "numpy",
+            "packaging",
+            "functorch @ git+https://github.com/pytorch/functorch@v0.2.0",
```

Comment:
We're installing functorch in install.sh but it's broken for some reason. I will fix this tonight if that's ok. In the meantime can you remove it from there?

Reply:
OFC. The tests won't pass for now though -- and I am not sure the `setup_evn.sh` update resolves all the `OSMesa` dependency cause I can't test it locally.

Reply:
BTW I think the reason is listed here:
https://pytorch.org/functorch/stable/install.html

```
functorch’s Linux binaries are compatible with all PyTorch 1.12.0 binaries aside from the PyTorch 1.12.0 cu102 binary; functorch will raise an error if it is used with an incompatible PyTorch binary. This is due to a bug in PyTorch ([pytorch/pytorch#80489](https://github.com/pytorch/pytorch/issues/80489)) and will be fixed in the next PyTorch minor release.

As a workaround, please install functorch from source: pip install --user git+https://github.com/pytorch/functorch@v0.2.0
```

Reply:
In theory we're using cuda 11.3
Bear with me I'll try to fix it later today!

Reply:
Take ur time :)

Reply:
It should be ok now!